src/test/java/hw4/ExerciseOneFunctionalityTest.java

```diff
@@ -0,0 +1,117 @@
+
+        //5. Assert Browser title
+        baseTest.assertString(mainPage.returnPageTitle(), testData.getPageTitle());
+
+        //6. Assert that there are 4 items on the header section displayed
+        // and they have proper texts
+        baseTest.assertList(mainPage.returnHeaderItemsText(), testData.getHeaderItemsNames());
+
+        //7.Assert that there are 4 images on the Index Page and they are displayed
+        baseTest.assertInt(mainPage.returnIconsNumber(), testData.getIconsNumber());
+        baseTest.assertTrue(mainPage.iconsAreDisplayed());
+
+        //8. Assert that there are 4 texts on the Index Page under icons and they have proper text
+        baseTest.assertInt(mainPage.returnBenefitTextsNumber(), testData.getIconsNumber());
+        baseTest.assertList(mainPage.returnBenefitTexts(), testData.getBenefitTexts());
+
+        //9. Assert a text of the main headers
+        baseTest.assertString(mainPage.returnMainTitleText(), testData.getMainTitle());
+        baseTest.assertString(mainPage.returnMainText(), testData.getMainText());
+
+        //10. Assert that there is the iframe in the center of page
+        baseTest.assertInt(mainPage.returnFramesNumber(), testData.getFramesNumber());
+
+        //11. Switch to the iframe and check that
+        // there is Epam logo in the left top conner of iframe
+        //12. Switch to original window back
+        baseTest.assertTrue(mainPage.isLogoInFrameExists());
+
+        //13. Assert a text of the sub header
+        baseTest.assertString(mainPage.returnSubHeaderText(), testData.getSubHeaderText());
+
+        //14. Assert that JDI GITHUB is a link and has a proper URL
+        baseTest.assertString(mainPage.returnSubHeaderUrl(), testData.getSubHeaderUrl());
+
+        //15. Assert that there is Left Section
+        baseTest.assertTrue(mainPage.isLeftSectionDisplayed());
+
+        //16. Assert that there is Footer
+        baseTest.assertTrue(mainPage.isFooterDisplayed());
+    }
+
+    private void initDriver() {
+        String propertyPath = "./src/main/resources/app.properties";
+        appProperties = new Properties();
+        try {
+            appProperties.load(new FileInputStream(propertyPath));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        String driverPath = appProperties.getProperty("driverPath");
+        System.out.println(driverPath);
+        System.setProperty("webdriver.chrome.driver", driverPath);
+        driver = new ChromeDriver();
+        driver.manage().window().maximize();
+    }
+
+    private void initMainPage() {
+        mainPage = new MainPage(driver);
+    }
+
+    private void initTestData() {
+        testData = new Ex1TestData();
+    }
+
+    private void initBaseTest() {
+        baseTest = new BaseTest();
+    }
+
+    @AfterClass
+    public void clearActions() {
+        //17. Close Browser
+        driver.quit();
+    }
```

Comment:
These methods can be moved to the parent class, looks like duplicated code fragment

Reply:
fixed